contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -148,6 +148,19 @@ contract OptimismPortal {
         emit TransactionDeposited(from, _to, msg.value, _value, _gasLimit, _isCreation, _data);
     }
 
+    /**
+     * @notice Determine if an L2 Output is finalized
+     * @param _l2Timestamp The timestamp of the L2 block
+     */
+
+    function isOutputFinalized(uint256 _l2Timestamp) external view returns (bool) {
```

Comment:
I think the OutputOracle would be a better location for this function. The OutputOracle should have the finalization window baked in because outputs cannot be deleted beyond the finalization window.

Reply:
Might require some other minor refactoring

Reply:
I agree that it makes more sense for the `OutputOracle` to have this function, but it doesn't have access to the finalization window. I initially wanted to add it there. I do think having this `view` function will be nice for frontends or off chain applications that want to know if an output has been finalized

Reply:
Hm. Actually this is a good question. Where the heck should this value live.

Reply:
When we refactor things to work off blocknumbers instead of timestamps, it could be a good time to introduce this change. I think it makes sense for the portal to hold the finalization window as its required for withdrawals and we don't want to have to call the output oracle to get it. The only other solution that i can think of is combining the output oracle and the portal but then thats a beefy contract and we will have to deal with its storage

Reply:
Word. I'll accept it for now.

Reply:
Let's use this function internally in `finalizeWithdrawalTransaction`

Reply:
If we use it internally, then we cannot return a bool, we would need to return `(bool, OutputProposal)` because the root is required downstream of the "is finalized" check, otherwise we add extra sloads

Reply:
This function is going to need to be refactored for https://github.com/ethereum-optimism/optimism/pull/2707 https://github.com/ethereum-optimism/optimism/pull/2703

We should expose this functionality for end users though, its very useful for offchain applications

Reply:
Those were dupes so 2707 is the only one that matters. Also, I'm pretty sure that  that the only change necessary is to modify the argument name:

```suggestion
    function isOutputFinalized(uint256 _l2BlockNumber) external view returns (bool) {
```

Reply:
I see the problem. Feel free to ignore my comment. Not worth the extra SLOADs.